tools/xabuild/XABuild.cs

```diff
@@ -30,57 +31,65 @@ static int Main ()
 			return MSBuildApp.Main ();
 		}
 
+		static EventWaitHandle waitHandle = new EventWaitHandle (true, EventResetMode.AutoReset, "XABUILD_SHARED_BY_ALL_PROCESSES");
```

Comment:
If I remember correctly EventWaitHandle on Mono doesn't implement cross-process sharing so this won't have any effect.

In general I think a better approach would be to open a FileStream with FileShare.Read arguments and catch the IOException if another process has the file open.

Reply:
Yeah this is the article we referred to at one point: http://aakinshin.net/blog/post/namedmutex-on-mono/

The PR is here: https://github.com/NuGet/NuGet.Client/pull/725

I'll see if I can come up with a PR.